src/daemon/utils.c

```diff
@@ -0,0 +1,15 @@
+#include "utils.h"
+#include "includes.h"
+
+void ckb_generate_random(unsigned int size, unsigned char* data) {
```

Comment:
Could directly return ssize_t to indicate success/failure (and also perhaps alert in case of short reads).